docs/lecture3.md

```diff
@@ -170,7 +173,120 @@ Note that the code below does not care about what are the fields of a post,
 because Firestore doesn't require you to have a predefined set of field. This
 gives you flexibility when writing your backend code.
 
-```javascript
+<Tabs
+groupId="lang"
+defaultValue="ts"
+values={[
+{ label: 'TypeScript', value: 'ts', },
+{ label: 'JavaScript', value: 'js', },
+]}>
+<TabItem value="ts">
+
+```typescript title="index.ts"
+// to use express with typescript be sure to `yarn add -D @types/express`
+// in typescript import packages as modules
+import admin from 'firebase-admin';
+import express from 'express'; // also install type aliases for Request, Response
+import { Request, Response } from 'express';
+import bodyParser from 'body-parser';
+
+const serviceAccount = require('./service-account.json');
+
+admin.initializeApp({
+  credential: admin.credential.cert(serviceAccount),
+  databaseURL: 'https://dti-web-dev-sp19-db-demo.firebaseio.com',
+});
+
+const db = admin.firestore();
+
+const app = express();
+const port: number = 8080;
+app.use(bodyParser.json());
+
+// Define a type for our Post document stored in Firebase
+type Post = {
+  content: string;
+  name: string;
+};
+
+type PostwithID = Post & {
+  id: string;
+};
+
+app.get('/', (req: Request, res: Response) => res.send('Hello World!'));
+
+const postsCollection = db.collection('posts');
+
+// create a post
+app.post('/post', function (req: Request, res: Response) {
```

Comment:
Don't really need to annotate here. TS can figure out the type from context (`app.post`'s type signature).